capabilities/src/execute_task_solution_capability.cpp

```diff
@@ -179,9 +179,10 @@ bool ExecuteTaskSolutionCapability::constructMotionPlan(const moveit_task_constr
 			for (auto& scene_diff : scene_diffs) {
 				if (!moveit::core::isEmpty(scene_diff)) {
 					/* RCLCPP_DEBUG_STREAM(LOGGER, "apply effect of " << description); */
-					scene_diff.robot_state.joint_state = sensor_msgs::msg::JointState();
-					scene_diff.robot_state.multi_dof_joint_state = sensor_msgs::msg::MultiDOFJointState();
-					if (!context_->planning_scene_monitor_->newPlanningSceneMessage(scene_diff))
+					auto scene_diff_copy = scene_diff;
+					scene_diff_copy.robot_state.joint_state = sensor_msgs::msg::JointState();
+					scene_diff_copy.robot_state.multi_dof_joint_state = sensor_msgs::msg::MultiDOFJointState();
+					if (!context_->planning_scene_monitor_->newPlanningSceneMessage(scene_diff_copy))
```

Comment:
This is a fixup for https://github.com/ros-planning/moveit_task_constructor/pull/461/commits/72afdd22d2a941350294c33fae3cccaf585f5b44 , please merge the commits.
Also, you should set `robot_state.is_diff = true` explicitly if you empty the robot_state fields and do it *before* you check for `isEmpty` because the set robot state will often be the only reason why the scene diff is not empty.

Could you split this feature into a separate PR? It's fully independent of the tiny other cleanups and might break something in the future.

Reply:
> This is a fixup for https://github.com/ros-planning/moveit_task_constructor/commit/72afdd22d2a941350294c33fae3cccaf585f5b44 , please merge the commits.

@v4hn Which commits are you referring to?